spec/services/place_hold_results/builder_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PlaceHoldResults::Builder do
+  subject(:builder) { described_class.new(user_token: user_token, client: symphony_client,
+                                          place_hold_results: place_hold_results) }
+
+  let(:processed_results) { builder.generate }
+  let(:user_token) { '2asf' }
+  let(:symphony_client) { instance_double(SymphonyClient) }
+  let(:hold_info_params) { ['a_hold_key', user_token] }
+  let(:item_info_params) { ['not_holdable_barcode', user_token] }
+  let(:parsed_hold_info) { JSON.parse HOLD_LOOKUP_RAW_JSON.to_json }
+  let(:parsed_item_info) { JSON.parse ITEM_LOOKUP_RAW_JSON.to_json }
+  let(:place_hold_results) { {
+    success: [{ barcode: 'holdable_barcode', hold_key: 'a_hold_key' }],
+    error: [{ barcode: 'not_holdable_barcode', error_message: 'User already has a hold on this material' }]
+  }.with_indifferent_access }
+
+  before do
+    allow(SymphonyClientParser).to receive(:parsed_response).with(symphony_client, :get_hold_info, *hold_info_params)
+      .and_return(parsed_hold_info)
+    allow(SymphonyClientParser).to receive(:parsed_response).with(symphony_client, :get_item_info, *item_info_params)
+      .and_return(parsed_item_info)
+  end
+
+  describe '#generate' do
+    it 'will pass along processed placed holds results' do
+      expect(processed_results[:success].first).to include(hold_key: 'a_hold_key')
+        .and include(barcode: 'holdable_barcode')
+        .and include(:placed_hold)
+    end
+
+    it 'returns Hold objects for placed holds' do
+      expect(processed_results[:success].first[:placed_hold]).to be_a Hold
```

Comment:
This spec and the one that starts on 49 use type checking. I kinda think it's good to avoid type checking, and maybe instead check for something unique about Hold. For example:

`expect(processed_results[:success].first[:placed_hold].pickup_library_human).to eq 'Pattee Library and Paterno Library Stacks'`

I think this makes the code more flexible, less brittle. If we ever renamed `Hold` this spec would start failing as it is, but if we just test against a unique characteristic of Hold (like the human pickup library name) it wouldn't.

For `Item` we could use `barcode` for example.

What do you think?

Reply:
I see your point, just trying to find uniqueness for Item 🤔 I am not sure if barcode is unique to Item only.

Reply:
ok, makes sense, I changed the Hold spec as you suggested and deleted the Item one since there is already another example checking the barcode. So, it should be good now

Reply:
Yup. I think this is good too. Thanks.